### ISSUE TITLE
Exported resource property with custom setter in tool script cannot be cleared by resource picker in editor

### ISSUE THREAD

**Issue by BenNicholls** [godotengine/godot#65916]
### Godot version

4.0.beta 1

### System information

Windows 10

### Issue description

I have an exported property for a texture with a setter function written that applies to texture to a mesh. The script is a tool script, so I can see the textures being applied to the relevant mesh in the editor. I want to delete the texture from the property and return it to its default null value. In godot 3.5, the Clear button in the resource picker did exactly this.

I expected the clear button to call the setter function of the property with a null value, but it appears that it does not call the setter at all and therefore no change is applied and the property is not cleared.

If there's something I'm misunderstanding here please let me know. This is something I did in Godot 3.5 but I know properties have changed a bit since then so maybe I've got something mixed up.

### Steps to reproduce

1. Open reproduction project
2. Select ResourcePickerExample node
3. Set example texture property with a new value, note in output that setter is called (statement will be printed).
4. Try to clear example texture property via the resource picker, note in output the setter is not called, and the texture remains assigned to the property.

### Minimal reproduction project

[ResourcePickerExample.zip](https://github.com/godotengine/godot/files/9586216/ResourcePickerExample.zip)

**Comment by nongvantinh** [comment]
Follow 3.5 codebase, after `Clear` clicked the following functions must be call:
```
>	godot.windows.tools.64.exe!PlaceHolderScriptInstance::set(const StringName & p_name, const Variant & p_value) Line 436	C++
 	godot.windows.tools.64.exe!Object::set(const StringName & p_name, const Variant & p_value, bool * r_valid) Line 412	C++
 	godot.windows.tools.64.exe!UndoRedo::_process_operation_list(List<UndoRedo::Operation,DefaultAllocator>::Element * E) Line 300	C++
 	godot.windows.tools.64.exe!UndoRedo::redo() Line 328	C++
 	godot.windows.tools.64.exe!UndoRedo::commit_action() Line 252	C++
 	godot.windows.tools.64.exe!EditorInspector::_edit_set(const String & p_name, const Variant & p_value, bool p_refresh_all, const String & p_changed_field) Line 2070	C++
 	godot.windows.tools.64.exe!EditorInspector::_property_changed(const String & p_path, const Variant & p_value, const String & p_name, bool changing) Line 2086	C++
 	godot.windows.tools.64.exe!MethodBind4<EditorInspector,String const &,Variant const &,String const &,bool>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 3153	C++
 	godot.windows.tools.64.exe!Object::call(const StringName & p_method, const Variant * * p_args, int p_argcount, Variant::CallError & r_error) Line 918	C++
 	godot.windows.tools.64.exe!Object::emit_signal(const StringName & p_name, const Variant * * p_args, int p_argcount) Line 1230	C++
 	godot.windows.tools.64.exe!EditorProperty::emit_changed(const StringName & p_property, const Variant & p_value, const StringName & p_field, bool p_changing) Line 110	C++
 	godot.windows.tools.64.exe!EditorPropertyResource::_resource_changed(const Ref<Resource> & p_resource) Line 2436	C++
 	godot.windows.tools.64.exe!MethodBind1<EditorPropertyResource,Ref<Resource> const &>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 849	C++
 	godot.windows.tools.64.exe!Object::call(const StringName & p_method, const Variant * * p_args, int p_argcount, Variant::CallError & r_error) Line 918	C++
 	godot.windows.tools.64.exe!Object::emit_signal(const StringName & p_name, const Variant * * p_args, int p_argcount) Line 1230	C++
 	godot.windows.tools.64.exe!Object::emit_signal(const StringName & p_name, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5, const Variant & p_arg6, const Variant & p_arg7, const Variant & p_arg8) Line 1286	C++
 	godot.windows.tools.64.exe!EditorResourcePicker::_edit_menu_cbk(int p_which) Line 276	C++
 	godot.windows.tools.64.exe!MethodBind1<EditorResourcePicker,int>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 849	C++
 	godot.windows.tools.64.exe!Object::call(const StringName & p_method, const Variant * * p_args, int p_argcount, Variant::CallError & r_error) Line 918	C++
 	godot.windows.tools.64.exe!Object::emit_signal(const StringName & p_name, const Variant * * p_args, int p_argcount) Line 1230	C++
 	godot.windows.tools.64.exe!Object::emit_signal(const StringName & p_name, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5, const Variant & p_arg6, const Variant & p_arg7, const Variant & p_arg8) Line 1286	C++
 	godot.windows.tools.64.exe!PopupMenu::activate_item(int p_item) Line 1168	C++
 	godot.windows.tools.64.exe!PopupMenu::_gui_input(const Ref<InputEvent> & p_event) Line 371	C++
 	godot.windows.tools.64.exe!MethodBind1<PopupMenu,Ref<InputEvent> const &>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 849	C++
 	godot.windows.tools.64.exe!Object::call_multilevel(const StringName & p_method, const Variant * * p_args, int p_argcount) Line 766	C++
 	godot.windows.tools.64.exe!Object::call_multilevel(const StringName & p_name, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5, const Variant & p_arg6, const Variant & p_arg7, const Variant & p_arg8) Line 864	C++
 	godot.windows.tools.64.exe!Viewport::_gui_call_input(Control * p_control, const Ref<InputEvent> & p_input) Line 1659	C++
 	godot.windows.tools.64.exe!Viewport::_gui_input_event(Ref<InputEvent> p_event) Line 2047	C++
 	godot.windows.tools.64.exe!Viewport::input(const Ref<InputEvent> & p_event) Line 2868	C++
 	godot.windows.tools.64.exe!Viewport::_vp_input(const Ref<InputEvent> & p_ev) Line 1425	C++
 	godot.windows.tools.64.exe!MethodBind1<Viewport,Ref<InputEvent> const &>::call(Object * p_object, const Variant * * p_args, int p_arg_count, Variant::CallError & r_error) Line 849	C++
 	godot.windows.tools.64.exe!Object::call(const StringName & p_method, const Variant * * p_args, int p_argcount, Variant::CallError & r_error) Line 918	C++
 	godot.windows.tools.64.exe!Object::call(const StringName & p_name, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5, const Variant & p_arg6, const Variant & p_arg7, const Variant & p_arg8) Line 847	C++
 	godot.windows.tools.64.exe!SceneTree::call_group_flags(unsigned int p_call_flags, const StringName & p_group, const StringName & p_function, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5, const Variant & p_arg6, const Variant & p_arg7, const Variant & p_arg8) Line 304	C++
 	godot.windows.tools.64.exe!SceneTree::input_event(const Ref<InputEvent> & p_event) Line 463	C++
 	godot.windows.tools.64.exe!InputDefault::_parse_input_event_impl(const Ref<InputEvent> & p_event, bool p_is_emulated) Line 491	C++
 	godot.windows.tools.64.exe!InputDefault::flush_buffered_events() Line 714	C++
 	godot.windows.tools.64.exe!OS_Windows::process_events() Line 2591	C++
 	godot.windows.tools.64.exe!OS_Windows::run() Line 3526	C++
 	godot.windows.tools.64.exe!widechar_main(int argc, wchar_t * * argv) Line 176	C++
 	godot.windows.tools.64.exe!_main() Line 198	C++
 	godot.windows.tools.64.exe!main(int _argc, char * * _argv) Line 210	C++
 	[External Code]	
```

But codebase with GDScript 2.0 in Godot 4.0 work very difference, I'm leaving this for anyone who familiar with `new GDScript` codebase.

**Comment by Dafaque** [comment]
Same here
Godot Engine v4.0.1.stable.mono.official.cacf49999

```
[Export]
public Skeleton3D _skeleton {
        set {
            GD.Print("CALL _skeleton.set");
            skeleton = value;
            NotifyPropertyListChanged();
        }
        get {return skeleton;}
    }
```
Message appear only when running game, but never in editor

```
--- Debugging process started ---
CALL _skeleton.set
Godot Engine v4.0.1.stable.mono.official.cacf49999 - https://godotengine.org
Vulkan API 1.3.236 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce MX150
 
CALL _skeleton.set
--- Debugging process stopped ---
_GetPropertyList() skeleton == null: 	True
_GetPropertyList() skeleton == null: 	True
Задать metadata/_editor_prop_ptr__skeleton
<Expect CALL _skeleton.set HERE>
Задать metadata/_editor_prop_ptr__skeleton
<Expect CALL _skeleton.set HERE>
```

**Comment by zvodd** [comment]
Mark this as **closed** I think.
The setter is being called both on clearing and setting the resource in the inspector - tested: 4.0.3 and 4.2.1
4.0.3 seems to call the setter twice on clear: resolved somewhere in between 4.0.3 and 4.2.1

**Comment by akien-mga** [comment]
Thanks for testing! Seems to work properly in 4.1+ already.
